File: testproject-servicio-financiero/src/main/java/com/testproject/app/servicio/service/ProductoService.java
package com.testproject.app.servicio.service;

import java.util.List;

import com.testproject.app.servicio.entity.Categoria;
import com.testproject.app.servicio.entity.Producto;

public interface ProductoService {
	
	public List<Producto> getProductos();
	public Producto getProducto(Long productoId);
	public Producto insertProducto(Producto producto);
	public Producto updateProducto(Producto producto);
	public Producto deleteProducto(Long productoId);
	public List<Producto> findByCategoria(Categoria categoria);	
	public List<Producto> findByCondicion(int edad, int residencia, double ingreso);

}

File: testproject-web/services/producto.Service.js
import fetchWrapper from '../helpers/fetch-wrapper';
const API_ENDPOINT = process.env.NEXT_PUBLIC_API_ENDPOINT;
//const api = "http://localhost:8091"
//const api = "http://host.docker.internal:8091"
const api = "http://host.docker.internal:9090/servicio-ws"
const APIPRODUCTO = `${api}/productos`;

const productoService = {
    getProductos: async() => {        
        return await (fetchWrapper.get(`${APIPRODUCTO}`));
    },
    getProductosByCondicion: async(edad, residencia, ingreso) => {        
        return await (fetchWrapper.get(`${APIPRODUCTO}/productosbycondicion?edad=${edad}&residencia=${residencia}&ingreso=${ingreso}`));
    },
    insertProducto: async(data) => {        
        return await (fetchWrapper.post(`${APIPRODUCTO}`, data));
    },
    updateProducto: async(data) => {        
        return await (fetchWrapper.put(`${APIPRODUCTO}`, data));
    },
    deleteProducto: async(id) => {
        return await (fetchWrapper.delete(`${APIPRODUCTO}/${id}`));
    }
}

export default productoService;
